math/src/main/java/com/powsybl/math/timeseries/TimeSeries.java

```diff
@@ -9,29 +9,284 @@
+                for (int j = 0; j < times.size(); j++) {
+                    stringValues.add(null);
+                }
+            }
+            return stringValues;
+        }
+
+        void parseToken(int i, String token) {
+            if (dataTypes[i - 2] == null) {
+                // test double parsing, in case of error we consider it a string time series
+                if (Doubles.tryParse(token) != null) {
+                    dataTypes[i - 2] = TimeSeriesDataType.DOUBLE;
+                    TDoubleArrayList doubleValues = createDoubleValues();
+                    doubleValues.add(parseDouble(token));
+                    values[i - 2] = doubleValues;
+                } else {
+                    dataTypes[i - 2] = TimeSeriesDataType.STRING;
+                    List<String> stringValues = createStringValues();
+                    stringValues.add(checkString(token));
+                    values[i - 2] = stringValues;
+                }
+            } else {
+                if (dataTypes[i - 2] == TimeSeriesDataType.DOUBLE) {
+                    ((TDoubleArrayList) values[i - 2]).add(parseDouble(token));
+                } else if (dataTypes[i - 2] == TimeSeriesDataType.STRING) {
+                    ((List<String>) values[i - 2]).add(checkString(token));
+                } else {
+                    throw assertDataType(dataTypes[i - 2]);
+                }
+            }
+        }
+
+        void parseLine(String[] tokens) {
+            for (int i = 2; i < tokens.length; i++) {
+                String token = tokens[i].trim();
+                parseToken(i, token);
+            }
+            // empty last cell case
+            if (tokens.length == names.size() + 1) {
+                parseToken(tokens.length, "");
+            }
+
+            times.add(ZonedDateTime.parse(tokens[0]));
+        }
+
+        void reInit() {
+            // re-init
+            times.clear();
+            for (int i = 0; i < dataTypes.length; i++) {
+                if (dataTypes[i] == TimeSeriesDataType.DOUBLE) {
+                    ((TDoubleArrayList) values[i]).clear();
+                } else if (dataTypes[i] == TimeSeriesDataType.STRING) {
+                    ((List) values[i]).clear();
+                } else {
+                    throw assertDataType(dataTypes[i]);
+                }
+            }
+        }
+
+        private Duration checkRegularSpacing() {
+            if (times.size() < 2) {
+                throw new TimeSeriesException("At least 2 rows are expected");
+            }
+
+            Duration spacing = null;
+            for (int i = 1; i < times.size(); i++) {
+                Duration duration = Duration.between(times.get(i - 1), times.get(i));
+                if (spacing == null) {
+                    spacing = duration;
+                } else {
+                    if (!duration.equals(spacing)) {
+                        throw new TimeSeriesException("Time spacing is expected to regular");
```

Comment:
throw new TimeSeriesException("Time spacing has to be regular")

Reply:
Done